erp.py

```diff
@@ -64,11 +74,19 @@ def wrapped_func(*args, **kwargs):
             'requestedUrl': 'https://erp.iitkgp.ac.in/IIT_ERP3/welcome.jsp',
         }
 
-
         r = s.post(ERP_LOGIN_URL, data=login_details,
                    **req_args)
+
+        if len(r.history) < 2:
+            print("{answer} (ERP_A{index}) is wrong for {question}".format(
+                answer=secret_answer,
+                index=secret_answer_index,
+                question=secret_question
+            ))
+            raise SecretAnswerError
```

Comment:
The secret answer error gets triggered even if the main password is itself wrong. Also, maybe add a check for if the roll number is valid or not (Check if secret question exists or not).

Reply:
Faced this issue yesterday while setting it up locally. Didn't realize this until I recheck my main password   :+1: